crates/uv/src/commands/tool/ensurepath.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::Write;
+
+use anyhow::Result;
+use owo_colors::OwoColorize;
+use tokio::io::AsyncWriteExt;
+use tracing::debug;
+
+use uv_configuration::PreviewMode;
+use uv_fs::Simplified;
+use uv_tool::find_executable_directory;
+use uv_warnings::warn_user_once;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+use crate::shell::Shell;
+
+/// List installed tools.
+pub(crate) async fn ensurepath(preview: PreviewMode, printer: Printer) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user_once!("`uv tool ensurepath` is experimental and may change without warning.");
+    }
+
+    let executable_directory = find_executable_directory()?;
+    debug!(
+        "Ensuring that the executable directory is in PATH: {}",
+        executable_directory.simplified_display()
+    );
+
+    if Shell::contains_path(&executable_directory) {
+        writeln!(
+            printer.stderr(),
+            "Executable directory {} is already in PATH",
+            executable_directory.simplified_display().cyan()
+        )?;
+    } else {
+        // Determine the current shell.
+        let Some(shell) = Shell::from_env() else {
+            return Err(anyhow::anyhow!("The executable directory {} is not in PATH, but the current shell could not be determined.", executable_directory.simplified_display().cyan()));
+        };
+
+        // Look up the configuration files (e.g., `.bashrc`, `.zshrc`) for the shell.
+        let files = shell.configuration_files();
+        if files.is_empty() {
+            return Err(anyhow::anyhow!("The executable directory {} is not in PATH, but updating {shell} is currently unsupported.", executable_directory.simplified_display().cyan()));
+        }
+
+        // Prepare the command (e.g., `export PATH="$HOME/.cargo/bin:$PATH"`).
+        let Some(command) = shell.prepend_path(&executable_directory) else {
+            return Err(anyhow::anyhow!("The executable directory {} is not in PATH, but the necessary command to update {shell} could not be determined.", executable_directory.simplified_display().cyan()));
+        };
```

Comment:
There's also the rustup pattern of a `~/.cargo/env` file e.g.

```
❯ cat ~/.cargo/env
#!/bin/sh
# rustup shell setup
# affix colons on either side of $PATH to simplify matching
case ":${PATH}:" in
    *:"$HOME/.cargo/bin":*)
        ;;
    *)
        # Prepending path in case a system-installed rustc needs to be overridden
        export PATH="$HOME/.cargo/bin:$PATH"
        ;;
esac
```

then we can suggest `source` and add that to the shell instead? Idk what the trade-offs are there.

Reply:
Yes I looked at that and the implementation, but I've never seen it before in other tools.